### PR TITLE
Refactor error handling

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
Replaces the depreciated `raven-js` lib with the latest version of `@Sentry/browser`.

- Fixes issue with the config of the previous implementation to ensure errors are captured.
- Maintains feature parity (with 1 exception regarding dataCallback - see below)
- Simplifies code

### I removed the `dataCallback` function
As far as I could tell, this code is redundant but please **carefully review** here to ensure this is the case.
```
// This function was being run for each error event to enhance the data
// we send to Sentry
const dataCallback: (data: {	
    culprit: string | undefined;	
    tags: {	
        origin: string;	
    };	
}) => {} = data => {	
    const { culprit } = data;	
    const resp = Object.assign({}, data);	

     if (culprit) {	
        const culpritMatches = /j.ophan.co.uk/.test(culprit);	
        resp.tags.origin = culpritMatches ? 'ophan' : 'dotcom-rendering';	
        resp.culprit = culprit.replace(/\/[a-z\d]{32}(\/[^/]+)$/, '$1');	
    }	

     return resp;	
};
```

## Changes in bundles size
This change increases the raven > sentry bundle size from 11.5kb to 18.5kb - which is a [known issue](https://github.com/getsentry/sentry-javascript/issues/1552). 7kb isn't great and I think we should look at using code splitting to only load this lib when it is required.

### Before
image

### After
image

## Link to supporting Trello card
https://trello.com/c/Btod8Mbz/755-refactor-client-side-error-handling

### CODE REVIEW

File: packages/frontend/web/browser/sentry/sentry.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    const { editionLongForm, contentType } = window.guardian.app.data.CAPI;
+    const {
+        isDev,
+        switches: { enableSentryReporting },
+    } = window.guardian.app.data.config;
+
+    Sentry.init({
+        ignoreErrors,
+        whitelistUrls,
+        dsn: 'https://1937ab71c8804b2b8438178dfdd6468f@sentry.io/1377847',
```
Comment (SiAdcock) [inline packages/frontend/web/browser/sentry/sentry.tsx]: Should this be hardcoded? I think [previously](https://github.com/guardian/dotcom-rendering/pull/831/files#diff-85c4844add4bd4441488aec0fff17145L78-L80) it was coming from data on the `window`

File: packages/frontend/web/browser/sentry/sentry.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        },
+    });
+
+    Sentry.configureScope(scope => {
+        scope.setTag('edition', editionLongForm);
+        scope.setTag('contentType', contentType);
+    });
+};
+
+export const reportError = (error: any, feature?: string): void => {
```
Comment (SiAdcock) [inline packages/frontend/web/browser/sentry/sentry.tsx]: Can `error` be typed as `Error`?

Comment (oliverlloyd) [reply packages/frontend/web/browser/sentry/sentry.tsx]: Yep. It should have been, thanks for spotting that.

File: packages/frontend/web/browser/sentry/sentry.tsx (hunk 1 above)
Comment (oliverlloyd) [reply packages/frontend/web/browser/sentry/sentry.tsx]: This is now done.

**Pull request merged**